src/esphome/binary_sensor/ttp229_lsf_sensor.cpp

```diff
@@ -0,0 +1,96 @@
+#include "esphome/defines.h"
+#ifdef USE_TTP229_LSF
+
+#include "esphome/binary_sensor/ttp229_lsf_sensor.h"
+#include "esphome/log.h"
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace binary_sensor {
+
+static const char *TAG = "binary_sensor.ttp229";
+
+TTP229Channel::TTP229Channel(const std::string &name, int channel_num) : BinarySensor(name) { channel_ = channel_num; }
+
+void TTP229Channel::process(const uint16_t *data, const uint16_t *last_data) {
+  if ((*data & (1 << this->channel_)) && !(*last_data & (1 << this->channel_))) {
+    this->publish_state(true);
+  }
+  if (!(*data & (1 << this->channel_)) && (*last_data & (1 << this->channel_))) {
+    this->publish_state(false);
+  }
+}
+
+TTP229LSFComponent::TTP229LSFComponent(I2CComponent *parent, uint8_t address) : I2CDevice(parent, address) {}
+
+void TTP229LSFComponent::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up ttp229...");
+  if (!this->parent_->raw_request_from(this->address_, 2)) {
+    this->error_code_ = COMMUNICATION_FAILED;
+    this->mark_failed();
+    return;
+  }
+}
+
+void TTP229LSFComponent::dump_config() {
+  ESP_LOGCONFIG(TAG, "ttp229:");
+  LOG_I2C_DEVICE(this);
+  switch (this->error_code_) {
+    case COMMUNICATION_FAILED:
+      ESP_LOGE(TAG, "Communication with ttp229 failed!");
+      break;
+    case WRONG_CHIP_STATE:
+      ESP_LOGE(TAG, "ttp229 has wrong default value for CONFIG2?");
+      break;
+    case NONE:
+    default:
+      break;
+  }
+}
+
+float TTP229LSFComponent::get_setup_priority() const { return setup_priority::HARDWARE_LATE; }
+
+TTP229Channel *TTP229LSFComponent::add_channel(binary_sensor::TTP229Channel *channel) {
+  this->channels_.push_back(channel);
+  return channel;
+}
+
+void TTP229LSFComponent::process_(uint16_t *data, uint16_t *last_data) {
+  for (auto *channel : this->channels_) {
+    channel->process(data, last_data);
+  }
+}
+
+uint8_t TTP229LSFComponent::byte_reverse_(uint8_t byte) {
```

Comment:
There's already a helper for this in `helpers.h` (that is also a bit more efficient, but that's not important).

https://esphome.io/api/helpers_8cpp.html#a96247e8095833e36a6510ed9737da079

Though in this case you probably want to use the 16bit variant of that

Reply:
this is so nice. I will remeber the helpers.h!